osquery/sql/virtual_table.cpp

```diff
@@ -1006,9 +1004,6 @@ static int xFilter(sqlite3_vtab_cursor* pVtabCursor,
       LOG(ERROR) << "Exception while executing table " << pVtab->content->name
                  << ": " << e.what();
       setTableErrorMessage(pVtabCursor->pVtab, e.what());
-      if (FLAGS_table_exceptions) {
-        throw;
```

Comment:
Can we maybe just remove this test but leave this flag and feature in place?

This flag is helpful when debugging exceptions thrown in tables. Assuming you are debugging an exception that was logged via the `LOG(ERROR)` line above, do you have a better way to find where in table code an exception is being thrown?

Reply:
I don't have a strong opinion in this instances. Does it make sense (and is it even possible) to always log the stack trace here without the flag logic?

Reply:
I have a strong opinion to keep it because I don't know a better way to allow me to quickly root cause people's issues.

For example consider the issue here: https://github.com/osquery/osquery/issues/7031

If I am able to reproduce the issue I then need to find the exact spot in code where the initial exception is being thrown. Without the `FLAGS_table_exceptions` I have to modify this broad catch statement to rethrow the exception.

Reply:
That's why I was wondering about just logging the stacktrace everytime.

Reply:
Just to understand, what were you expecting to see there/how you were debugging?
 A re-thrown exception that's being caught by a debugger, won't show the original throw location, but the stack trace will start where the last throw is, which means, the one in the catch.
To get the original throw you would have to do `catch throw` in gdb, so that then you can print the stack trace of all throws and find the one at the origin.
 If it's a matter of getting the message the exception was carrying, then we already have it in the table error message.

Reply:
For context, I marked this as resolved since the commit changing the code was reverted and the test was fixed with an update to the CMake files